foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/descriptors/ClassDescriptor.java

```diff
@@ -1368,7 +1369,7 @@ public Object clone() {
         clonedDescriptor.setFields(NonSynchronizedVector.newInstance());
 
         // Referencing classes
-        referencingClasses = new HashSet<>();
+        referencingClasses = new ConcurrentHashMap<>().newKeySet();
```

Comment:
this the place which is wrong and the cause of the `ConcurrentModificationException`. This line should be `clonedDescriptor.referencingClasses = new HashSet<>(referencingClasses);`